lib/ecto/adapters/postgres/connection.ex

```diff
@@ -414,6 +414,14 @@ if Code.ensure_loaded?(Postgrex) do
     defp operator_to_boolean(:and), do: " AND "
     defp operator_to_boolean(:or), do: " OR "
 
+    defp parens_for_select([first_expr | _] = expr) do
+      if String.match?(first_expr, ~r/SELECT/i) do
```

Comment:
Let's go with: `is_binary(first_expr) and String.starts_with?(first_expr, ["SELECT", "select"])`

That's because the first expression is not always a binary and binary matches are much faster than regexes.

Reply:
Sounds good. I take it we are not worried about whitespace accidentally left at the beginning of the string then?